api/v1beta1/machine_types.go

```diff
@@ -230,10 +230,32 @@ type MachineStatus struct {
 	// Conditions defines current service state of the Machine.
 	// +optional
 	Conditions Conditions `json:"conditions,omitempty"`
+
+	// Deletion contains information relating to removal of the Machine.
+	// Only present when the Machine has a deletionTimestamp and is being removed from the cluster.
+	// +optional
+	Deletion MachineStatusDeletion `json:"deletion,omitempty"`
 }
 
 // ANCHOR_END: MachineStatus
 
+// MachineStatusDeletion is the deletion state of the Machine.
+type MachineStatusDeletion struct {
+	// NodeDrainStartTime is the time when the drain of the node started.
+	// Only present when the Machine has a deletionTimestamp, is being removed from the cluster
+	// and draining the node had been started.
```

Comment:
Why would drain not have been started when there's a deletion timestamp?

Reply:
Because of waiting for pre-drain hooks is one thing.

Reply:
Another one may be the node is excluded from the drain entirely (there is an annotation for it).

I shortened it to:
```
	// NodeDrainStartTime is the time when the drain of the node started and is used to determine
	// if the NodeDrainTimeout is exceeded.
	// Only present when the Machine has a deletionTimestamp and draining the node had been started.
```

Happy to get other suggestions though :-)

One way would be to say:

```
	// NodeDrainStartTime is the time when the drain of the node started and is used to determine
	// if the NodeDrainTimeout is exceeded.
	// Only present when the Machine has a deletionTimestamp and draining the node had been started.
    // NodeDrainStartTime may not be set because the deletion is blocked by a pre-drain hook or draining is skipped for the machine.
```

But doing the same for `WaitForNodeVolumeDetachStartTime` would get very verbose 🤔  :

```
	// WaitForNodeVolumeDetachStartTime is the time when waiting for volume detachment started
	// and is used to determine if the NodeVolumeDetachTimeout is exceeded.
	// Detaching volumes from nodes is usually done by CSI implementations and the current state
	// is observed from the node's `.Status.VolumesAttached` field.
	// Only present when the Machine has a deletionTimestamp and waiting for volume detachments had been started.
    // WaitForNodeVolumeDetachStartTime may not be set because the deletion is blocked by a pre-drain hook, stuck in drain or waiting for volume detachment is skipped for the machine.
    ```